rpc/rpc.go

```diff
@@ -157,6 +157,12 @@ func (l *Link) RPC(ctx context.Context, msg *amqp.Message) (*Response, error) {
 	}
 	msg.Properties.ReplyTo = l.clientAddress
 
```

Comment:
You'll need a nil-check here on ApplicationProperties. While it should never be nil here, because folks need to provide the "operation" field in ApplicationProperties, I'd prefer they get back a server-error instead of this library starting a panic.

Reply:
Ok. Got it. But what would happen if `ApplicationProperties` is nil?

Reply:
You gotta new one up :) IIRC, it'll look like this:

``` Go
msg.ApplicationProperties = make(map[string]interface{})
```

Reply:
I have made the changes.
But one question, what are the chances that `ApplicationProperties` will be nil. When I traced backwards, `msg` variable always has some data.

Reply:
Low, as I said eariler:
> it should never be nil here because folks need to provide the "operation" field in ApplicationProperties, I'd prefer they get back a server-error instead of this library starting a panic.